### PR TITLE
Add Preflight Checks

### PR DESCRIPTION
**Pull request by davidmreed**
We now preflight-check before installing NPSP: 

- Chatter must be turned on for all install plans
- Advanced Currency Management must be turned off for all plans
- NPSP must be installed to run the Reports & Dashboards plan
- NPSP must _not_ be installed to run the Trailhead installer.
- The Trailhead installer will show a warning for any org other than a Trailhead org.
- We check for more My Domain patterns using a new CumulusCI preflight check.

Installer for testing (*VPN required*): https://metadeploy-stg.herokuapp.com/products/npsp/e3fa32896e37e979e7e871f2bcea388696695443/install

This PR will require CumulusCI 3.25 (release date 2020-12-10)

# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: cumulusci.yml [hunk 1]
```diff
@@ -1244,6 +1250,15 @@ plans:
...
               message: "Please enable My Domain in your org prior to installing."
+            - when: "not tasks.check_chatter_enabled()"
+              action: error
+              message: "Nonprofit Success Pack requires Chatter. Please enable Chatter in your org and try again."
+            - when: "not 'npsp' in org_config.installed_packages"
+              action: error
+              message: "Nonprofit Success Pack is already installed in this org. This installer cannot be used with orgs that already have NPSP."
+            - when: "not tasks.check_advanced_currency_management()"
+              action: error
+              message: "Advanced Currency Management must be disabled to install Nonprofit Success Pack, but can be enabled after installation. See the Power of Us Hub for assistance."
```
Comment (melissabarber) [inline cumulusci.yml:1261]:
```suggestion
              message: "Advanced Currency Management must be disabled to install Nonprofit Success Pack, but can be enabled after installation."
```

File: metadeploy/labels_en.json [hunk 2]
```diff
@@ -169,6 +169,22 @@
         "Please enable My Domain in your org prior to installing.": {
             "message": "Please enable My Domain in your org prior to installing.",
             "description": "shown if validation fails"
+        },
+        "Nonprofit Success Pack requires Chatter. Please enable Chatter in your org and try again.": {
+            "message": "Nonprofit Success Pack requires Chatter. Please enable Chatter in your org and try again.",
+            "description": "shown if validation fails"
+        },
+        "Advanced Currency Management must be disabled to install Nonprofit Success Pack, but can be enabled after installation. See the Power of Us Hub for assistance.": {
+            "message": "Advanced Currency Management must be disabled to install Nonprofit Success Pack, but can be enabled after installation. See the Power of Us Hub for assistance.",
```
Comment (melissabarber) [inline metadeploy/labels_en.json:178]:
```suggestion
            "message": "Advanced Currency Management must be disabled to install Nonprofit Success Pack, but can be enabled after installation.",
```

**Review by melissabarber — CHANGES_REQUESTED** [review]
Added the same suggestion a few times and added a comment about why I recommended that change. If it seems important to keep that sentence, maybe update it to, "See the Power of Us Hub for more information."

File: cumulusci.yml [hunk 3]
```diff
@@ -1208,6 +1208,12 @@ plans:
             - when: "'.my.' not in org_config.instance_url"
               action: error
               message: "Please enable My Domain in your org prior to installing."
+            - when: "not tasks.check_chatter_enabled()"
+              action: error
+              message: "Nonprofit Success Pack requires Chatter. Please enable Chatter in your org and try again."
+            - when: "not tasks.check_advanced_currency_management()"
+              action: error
+              message: "Advanced Currency Management must be disabled to install Nonprofit Success Pack, but can be enabled after installation. See the Power of Us Hub for assistance."
```
[the remark this answers is not in the document]
Comment (davidmreed) [reply cumulusci.yml:1216]: I don't think it's important. I'll adopt all of your suggestions.

**Review by melissabarber — APPROVED** [review]
Messaging looks great to me!

File: cumulusci.yml [hunk 4]
```diff
@@ -1244,6 +1250,15 @@ plans:
             - when: "'.my.' not in org_config.instance_url"
               action: error
               message: "Please enable My Domain in your org prior to installing."
+            - when: "not tasks.check_chatter_enabled()"
+              action: error
+              message: "Nonprofit Success Pack requires Chatter. Please enable Chatter in your org and try again."
+            - when: "not 'npsp' in org_config.installed_packages"
```
Comment (davidmreed) [inline cumulusci.yml:1256]:
```suggestion
            - when: "'npsp' in org_config.installed_packages"
```

File: cumulusci.yml [hunk 5]
```diff
@@ -1316,6 +1331,10 @@ plans:
         title: Install Reports & Dashboards
         tier: additional
         is_listed: True
+        checks:
+            - when: "not 'npsp' in org_config.installed_packages"
```
Comment (davidmreed) [inline cumulusci.yml:1335]:
```suggestion
            - when: "'npsp' not in org_config.installed_packages"
```

File: cumulusci.yml [hunk 6]
```diff
@@ -1208,6 +1208,12 @@ plans:
             - when: "'.my.' not in org_config.instance_url"
               action: error
               message: "Please enable My Domain in your org prior to installing."
+            - when: "not tasks.check_chatter_enabled()"
+              action: error
+              message: "Nonprofit Success Pack requires Chatter. Please enable Chatter in your org and try again."
+            - when: "not tasks.check_advanced_currency_management()"
```
Comment (davidmreed) [inline cumulusci.yml:1214]:
```suggestion
            - when: "tasks.check_advanced_currency_management()"
```

File: cumulusci.yml [hunk 7]
```diff
@@ -1244,6 +1250,15 @@ plans:
             - when: "'.my.' not in org_config.instance_url"
               action: error
               message: "Please enable My Domain in your org prior to installing."
+            - when: "not tasks.check_chatter_enabled()"
+              action: error
+              message: "Nonprofit Success Pack requires Chatter. Please enable Chatter in your org and try again."
+            - when: "'npsp' in org_config.installed_packages"
+              action: error
+              message: "Nonprofit Success Pack is already installed in this org. This installer cannot be used with orgs that already have NPSP."
+            - when: "not tasks.check_advanced_currency_management()"
```
Comment (davidmreed) [inline cumulusci.yml:1259]:
```suggestion
            - when: "tasks.check_advanced_currency_management()"
```

**Review by bdvorachek — APPROVED** [review]

**Pull request merged**